Modules/md5module.c

```diff
@@ -72,6 +72,7 @@ static MD5object *
 newMD5object(MD5State * st)
 {
     MD5object *md5 = (MD5object *)PyObject_GC_New(MD5object, st->md5_type);
+    md5->lock = NULL;
```

Comment:
so the lock is lazily-initialized and only created if at some points someone requests hashing data past the threshold?

Reply:
yep.